### DISCUSSION TITLE
semi sequential printing with multi stacking

### DISCUSSION THREAD

**Discussion by lunartester** [supermerill/SuperSlicer#1747] [Q&A]
If I want to print with a single nozzle printer this part:

https://ibb.co/D5tkhPt

so I want to print the 4 gray towers as single object first, but the next layer up the top red spheres 8 of them 2 on each top of each gray tower I want each pair of spheres to print separately one by one sequentially. so there is no stringing. I'm unable to do that, they won't stack up, and drop to the ground. Is there a solution where you can print few parts simultaneously up to certain height then stack up parts on top and continue printing them individually from start to finish but without any pause sicne it's single material printing. 

Cheers.

**Comment by supermerill** [comment]
>  you can print a few parts **simultaneously**
> then stack up parts on top [??]
> and **continue** printing them **individually**

Sorry but I don't understand.

**Comment by lunartester** [comment]
ok, lets say I import 4 towers up to 10mm height they will start to print simultaneously so each layer by layer is printed on all 4 towers, once I reach the top or lets say once i reach certain layer height, can I instruct the slicer from that point on not to print 4 towers layer by layer, but finish printing each tower individually, obviously this requires to set the "complete individual objects" option and additional set up of the nozzle clearance so it won't hit one tower after moving to another. Is there anyway to switch from default option which is what ever objects are on the plate they are printed all together but I can import more objects stack them on top and print then from that point on individually.

**Reply by supermerill** [reply]
oh, understood.
It's possible, but not before the gcodewriter rewrite.
For now, you can print them one after the other, no? (or put them at each corner fo the build plate)

**Reply by lunartester** [reply]
I don't know anything about programing or Gcodewriter, in basic terms how many many hours will it take you to implement this feature, I realize it;s not on priority list, but it's very important feature when ti comes to printing with filament that strings a lot, specially flexibles. I cannot print then one after another, I'm stuck with either printing each full tower with a spheres on top of it one after another or it all prints as one.  With the approach I offered I can maintain 40mm/sec speed even on very flexible filament, so if I have a full bed tray of these it's a big difference in printing for 36 hours vs 12 hours.

**Reply by supermerill** [reply]
many days. maybe less if done poorly.

What you 'can' do (very hacky):

create two gcode, 
* one for the first part
* the second with "complete object"

To be sure everything is at the same place, don't cut, and don't move them
Set an extruder height clearance of a meter, so it won't stop you slicing the second file.
Then, you have to open the gcode files. On the first, find the:
 ```
;LAYER_CHANGE
;Z:5.9
;HEIGHT:0.15
```
with the height of when you want to change, and remove everything after that.

Then on the other file, find the same gcode, and remove everything before. Then you have to find the travel to Z:0, and a second occurrence of the wanted height. And remove this portion, and again for every other tower you have.

And then assemble the two file, copying the second at the end of the first. It should be printable.

**Reply by lunartester** [reply]
Sorry this approach does not work for me, I always have different parts stacked on top of each other, doing it the way you offered each time is not optimal. Is there some type of script or something else that can be done to combine these tow Gcodes into one and change the lines as you described it?    

Option number two:

If I create two files one with spheres on to one without like this:

https://ibb.co/FXmPdtZ

but they are aligned of course, if I set the "Complete Individual Objects" extruder clearance to zero it still is an error, but if you can somehow override that error, maybe it will think that two objects have the same geometry up to the top end of the tower and once it reaches that top, it will not try to start printing from the very bottom the second part which is what the error shows, but it will think that the same geometry has been printed already and lets continue with the spheres, is there  a way for you to override it somehow.... If that its possible then that overridden version can only work for cases like mine, it cannot be a universal version for people that want to use the default setting of printing objects one by one.

**Reply by supermerill** [reply]
> Is there some type of script
it's possible to write a script, but I won't do that for you on my free time.

Btw, I wonder how you imagine printing these with "complete object", as they are too close to each other, they will be knocked by the extruder if more than two milimiter high.

**Reply by lunartester** [reply]
You are correct, these are very small parts, about 3mm, so they do clear the nozzle, if they were big and tall they would never clear, that's the whole objective, with this approach I can fit 5 times more parts on one built plate and leave it unattended vs doing 2-3 built plates with parts being spaced far a part. This is not a one off it's for production scaling approach.

**Reply by lunartester** [reply]
Please contact me privately if this is not a free implementation.
